core/Pos/Binary/Core/Coin.hs

```diff
@@ -115,20 +120,25 @@ hdrToParam h
     | isClear h 4 = (3, fromIntegral (h .&. 0x0f))
     | otherwise   = (4, fromIntegral (h .&. 0x0f))
 
-decodeVarint :: Peek Word64
-decodeVarint = do
-    (nbBytes, acc) <- hdrToParam <$> getWord8
-    conts <- replicateM nbBytes getWord8
-    let val = orAndShift acc conts
-    when (expectedContBytes val /= length conts) $ fail "not canonical encoding"
-    return val
+decodeMegaMicros :: [Word8] -> Decoder s (Word64, [Word8])
```

Comment:
I already wrote it in another comment but I'll repeat it here

> If you use decimal-fraction encoding, the difference will be around 2× (I checked), which is acceptable. Otherwise, common values like one ADA will be encoded as 1000000 coins = 1A 000F4240 = 5 bytes, which is kinda worse and I'd rather prefer decimal-fraction encoding here.

Basically, since Coin = 1e-6 ADA, it makes more sense to treat it as a decimal fraction because then we'll have optimal encoding for common cases like “1 ADA”, “1.5 ADA”, etc.